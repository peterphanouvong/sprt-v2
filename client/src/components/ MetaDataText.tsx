import { Text } from "@chakra-ui/react";
import React from "react";

interface Props {
  display?: string;
}

const MetaDataText: React.FC<Props> = ({ children, display = "inline" }) => {
  return (
    <Text display={display} color="GrayText" fontSize="sm">
      {children}
    </Text>
  );
};

export { MetaDataText };
