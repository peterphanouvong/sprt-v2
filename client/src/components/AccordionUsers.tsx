import { CheckIcon } from "@chakra-ui/icons";
import {
  Box,
  Accordion,
  AccordionItem,
  AccordionButton,
  AccordionIcon,
  AccordionPanel,
  ListItem,
  Text,
} from "@chakra-ui/react";
import React from "react";
import { User } from "../generated/graphql";

interface Props {
  userType: string;
  userList: User[];
  numHighlighted?: number;
}

const AccordionUsers: React.FC<Props> = ({
  userType,
  userList,
  numHighlighted = 0,
}) => {
  return (
    <Box mt={4}>
      <Accordion allowToggle>
        <AccordionItem>
          <AccordionButton
            _hover={{
              backgroundColor: "gray.50",
            }}
          >
            <Box flex='1' textAlign='left'>
              <Text fontWeight='medium'>{userType}</Text>
            </Box>
            <AccordionIcon />
          </AccordionButton>
          <AccordionPanel p={0}>
            {userList.map((attendee, index) => {
              const isHighlighted = index < numHighlighted;
              return (
                <ListItem
                  paddingY={1}
                  paddingX={4}
                  listStyleType='none'
                  key={attendee.id}
                  background={isHighlighted ? "green.50" : "yellow.50"}
                  _hover={{
                    backgroundColor: isHighlighted ? "green.100" : "yellow.100",
                  }}
                >
                  <Box
                    fontSize='sm'
                    color={isHighlighted ? "green.900" : "yellow.900"}
                    display='flex'
                    justifyContent='space-between'
                    alignItems='center'
                  >
                    {attendee.username}
                    {isHighlighted ? (
                      <CheckIcon color='green' />
                    ) : (
                      <Text>(waitlist)</Text>
                    )}
                  </Box>
                </ListItem>
              );
            })}
          </AccordionPanel>
        </AccordionItem>
      </Accordion>
    </Box>
  );
};

export { AccordionUsers };
