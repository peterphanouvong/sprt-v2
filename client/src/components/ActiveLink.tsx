import React from "react";
import NextLink from "next/link";
import { Box } from "@chakra-ui/layout";
import { useRouter } from "next/router";

interface Props {
  href: string;
}

const ActiveLink: React.FC<Props> = ({ children, href }) => {
  const router = useRouter();
  const isActive = router.pathname === href;
  return (
    <Box
      _hover={{ borderBottom: "1px solid", borderColor: "gray.300" }}
      borderBottom={isActive ? "1px solid" : ""}
      borderColor={isActive ? "gray.300" : ""}
    >
      <NextLink href={href}>
        <a>{children}</a>
      </NextLink>
    </Box>
  );
};

export { ActiveLink };
