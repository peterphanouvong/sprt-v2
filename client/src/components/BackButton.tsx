import { ChevronLeftIcon } from "@chakra-ui/icons";
import { ButtonProps, IconButton } from "@chakra-ui/react";
import { useRouter } from "next/router";
import React from "react";

type Props = ButtonProps;

const BackButton: React.FC<Props> = ({ ...props }) => {
  const router = useRouter();
  return (
    <IconButton
      isRound={true}
      aria-label="back-button"
      icon={<ChevronLeftIcon />}
      onClick={() => router.back()}
      {...props}
    />
  );
};

export { BackButton };
