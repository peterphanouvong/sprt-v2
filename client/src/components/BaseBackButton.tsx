import { ChevronLeftIcon } from "@chakra-ui/icons";
import { ButtonProps, IconButton } from "@chakra-ui/react";
import { useRouter } from "next/router";
import React from "react";

type Props = ButtonProps;

const BaseBackButton: React.FC<Props> = ({ ...props }) => {
  const router = useRouter();
  return (
    <IconButton
      isRound={true}
      colorScheme="gray"
      variant="solid"
      aria-label="back-button"
      icon={<ChevronLeftIcon />}
      onClick={() => router.back()}
      {...props}
    />
  );
};

export { BaseBackButton };
