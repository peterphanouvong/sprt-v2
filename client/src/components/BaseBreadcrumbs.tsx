import { ChevronRightIcon } from "@chakra-ui/icons";
import { Breadcrumb, BreadcrumbItem, BreadcrumbLink } from "@chakra-ui/react";
import React from "react";
import NextLink from "next/link";
import { Text } from "@chakra-ui/layout";

interface Props {
  crumbs: {
    title: string;
    href: string;
  }[];
}

const BaseBreadcrumbs: React.FC<Props> = ({ crumbs }) => {
  return (
    <Breadcrumb
      mb={4}
      spacing="8px"
      separator={<ChevronRightIcon color="gray.500" />}
    >
      {crumbs.map((crumb, index) => (
        <BreadcrumbItem key={index}>
          <BreadcrumbLink>
            <NextLink href={crumb.href}>
              <Text variant="body-3">{crumb.title}</Text>
            </NextLink>
          </BreadcrumbLink>
        </BreadcrumbItem>
      ))}
    </Breadcrumb>
  );
};

export { BaseBreadcrumbs };
