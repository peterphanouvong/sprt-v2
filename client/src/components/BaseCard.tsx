import { Box, BoxProps } from "@chakra-ui/react";
import React from "react";

type Props = BoxProps;

const BaseCard: React.FC<Props> = ({ ...props }) => {
  return (
    <Box
      borderRadius="sm"
      border="1px solid"
      bgColor="gray.50"
      borderColor="gray.200"
      padding={2}
      {...props}
    ></Box>
  );
};

export { BaseCard };
