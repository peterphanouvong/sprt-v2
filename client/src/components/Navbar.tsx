import React from "react";
import NextLink from "next/link";
import {
  Box,
  Button,
  HStack,
  Link,
  useBreakpointValue,
  useColorMode,
} from "@chakra-ui/react";

import Logo from "./Logo";
import { useLogoutMutation, useMeQuery } from "../generated/graphql";
import SettingsDrawer from "./SettingsDrawer";
import { EventCreateButton } from "./EventCreateButton";

interface Props {}

const Navbar: React.FC<Props> = ({}) => {
  const { colorMode } = useColorMode();
  const [{ fetching: logoutFetching }, logout] = useLogoutMutation();
  const [{ data, fetching }] = useMeQuery({});

  const hidden = useBreakpointValue({ base: true, sm: true, md: false });

  let body: any = null;
  let left: any = null;

  //data loading d
  if (fetching) {
    // user not logged in
  } else if (!data?.me) {
    body = (
      <HStack spacing={8}>
        <NextLink href="/login">
          <Link>login</Link>
        </NextLink>
        <NextLink href="/register">
          <Link>register</Link>
        </NextLink>
        <SettingsDrawer />
      </HStack>
    );

    // user logged in
  } else {
    body = (
      <HStack spacing={8}>
        <NextLink href={`/${data.me.username}`}>
          <Link>{data.me.username}</Link>
        </NextLink>
        <Button
          onClick={() => {
            logout();
          }}
          isLoading={logoutFetching}
          variant="link"
          fontWeight="normal"
        >
          logout
        </Button>
        <SettingsDrawer />
      </HStack>
    );

    left = (
      <HStack spacing={8}>
        <NextLink href="/feed">
          <Link>feed</Link>
        </NextLink>
        <NextLink href="/clubs">
          <Link>clubs</Link>
        </NextLink>{" "}
        <NextLink href="/events">
          <Link>events</Link>
        </NextLink>
        <EventCreateButton />
      </HStack>
    );
  }

  return (
    <Box
      paddingX={16}
      display="flex"
      justifyContent="space-between"
      alignItems="center"
      // borderBottomColor  ="gray.200"
      // boxShadow="sm"
      // borderBottomWidth="1px"
      mb={8}
      position="sticky"
      top={0}
      zIndex="10"
      bg={colorMode === "dark" ? "gray.800" : `white`}
      hidden={hidden}
    >
      <Box display="flex" alignItems="center">
        <NextLink href="/home">
          <a style={{ paddingTop: "6px" }}>
            <Logo color={colorMode === "dark" ? "white" : "black"} />
          </a>
        </NextLink>
        <Box mr={4} />
        {left}
      </Box>
      {body}
    </Box>
  );
};

export { Navbar };
