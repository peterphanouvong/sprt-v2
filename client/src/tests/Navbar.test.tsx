import { shallow } from "enzyme";
import { Navbar } from "../components/Navbar";

test("navbar snapshot", () => {
  expect(shallow(<Navbar />)).toMatchSnapshot();
});

// test("it shows login & register when not logged in");
// test("it shows username & logout when logged in");
