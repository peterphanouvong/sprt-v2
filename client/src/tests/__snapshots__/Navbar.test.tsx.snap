// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navbar snapshot 1`] = `
<Fragment>
  <Box
    alignItems="center"
    bg="white"
    display="flex"
    flexDirection="row"
    justifyContent="space-between"
    mb={8}
    paddingX={16}
    position="sticky"
    top={0}
    zIndex="10"
  >
    <Box
      alignItems="center"
      display="flex"
      flexDirection="row"
    >
      <Link
        href="/feed"
      >
        <a
          style={
            Object {
              "paddingTop": "6px",
            }
          }
        >
          <Logo
            color="black"
          />
        </a>
      </Link>
      <Box
        mr={4}
      />
    </Box>
  </Box>
</Fragment>
`;
